src/diffusers/models/attention.py

```diff
@@ -201,6 +220,80 @@ def forward(self, hidden_states, context=None):
         return hidden_states
 
 
+class MemoryEfficientCrossAttention(nn.Module):
+    def __init__(self, query_dim, context_dim=None, heads=8, dim_head=64, dropout=0.0):
+        super().__init__()
+        inner_dim = dim_head * heads
+        context_dim = default(context_dim, query_dim)
+
+        self.scale = dim_head**-0.5
+        self.heads = heads
+        self.dim_head = dim_head
+
+        self.to_q = nn.Linear(query_dim, inner_dim, bias=False)
+        self.to_k = nn.Linear(context_dim, inner_dim, bias=False)
+        self.to_v = nn.Linear(context_dim, inner_dim, bias=False)
+
+        self.to_out = nn.Sequential(nn.Linear(inner_dim, query_dim), nn.Dropout(dropout))
+        self.attention_op: Optional[Any] = None
+
+    def _maybe_init(self, x):
+        """
+        Initialize the attention operator, if required We expect the head dimension to be exposed here, meaning that x
+        : B, Head, Length
+        """
+        if self.attention_op is not None:
+            return
+
+        _, K, M = x.shape
```

Comment:
I believe the `K` / `M` are inverted here, which are causing issues with the kernels that are dispatched.

Reply:
Nice fix, that works for me - thanks! :D

Reply:
(I wonder why we didn't run into this before, lol)

Reply:
same, I'm a bit puzzled that this could work. I'm also very used to xformers dimensions being B,M,K so that's a bit strange that this slipped in and worked altogether. We'll update the PR to clean this up

Reply:
Just pushed a commit fixing this, thank you @fmassa